hibernate-core/src/main/java/org/hibernate/boot/model/naming/NamingHelper.java

```diff
@@ -133,10 +133,10 @@ public String generateHashedConstraintName(
 	 */
 	public String hashedName(String name) {
 		try {
-			final MessageDigest md5 = MessageDigest.getInstance( "MD5" );
-			md5.reset();
-			md5.update( charset != null ? name.getBytes( charset ) : name.getBytes() );
-			final BigInteger bigInt = new BigInteger( 1, md5.digest() );
+			final MessageDigest md = MessageDigest.getInstance( "SHA-256" );
+			md.reset();
+			md.update( charset != null ? name.getBytes( charset ) : name.getBytes() );
+			final BigInteger bigInt = new BigInteger( 1, md.digest() );
```

Comment:
We can't just **replace** the previous algorithm, since that would break schema migration for existing schemas.

What we could do is use SHA-256 iff MD5 is not available.

Reply:
I have used SHA-256 for hashing if MD5 is not available.